### DISCUSSION TITLE
New ligatures request [VERBOSE], # VERBOSE.

### DISCUSSION THREAD

**Discussion by ElonPark** [fabrizioschiavi/pragmatapro#195] [General]
Hello, first of all, thanks for the great fonts,
I am making and using a logger, in xcode
like this:
```swift
// default level string
enum Level: String {
    case verbose = "📢 [VERBOSE]"
    case debug = "🛠 [DEBUG]"
    case info = "💡 [INFO]"
    case warning = "⚠️ [WARNING]"
    case error = "🚨 [ERROR]"
}
```
and it looks like this:
<img width="497" alt="스크린샷 2019-12-02 오후 3 01 18" src="https://user-images.githubusercontent.com/13270453/69934198-9ade2380-1514-11ea-9515-66dd2f9f8362.png">

As you can see, only [VERBOSE] does not apply Rigature.
Can you add [VERBOSE], # VERBOSE like existing ligatures?

my logger here.
https://github.com/ElonPark/EPLogger

thanks for reading

**Comment by fabrizioschiavi** [comment]
Thanks @ElonPark for this request! 
I just added in my Todo list

**Comment by amerlyq** [comment]
I'm not against adding more ligatures, but @ElonPark , my extensive experience with embedded screams it's not the right thing to do for logging facility. Because you impose personal restrictions and quirks of only the temporary favor and importance onto something, which must be as simple and straightforward as possible. It may look nice for logs of screen worth but very tiring for longs in thousands and hundreds of thousands lines -- I talk both about ligatures and pictograms.
But whatever, try your best, maybe it will be worth it for at least some usecase.

**Comment by nomasprime** [comment]
These ligatures should be removed. Stopped using ligature variant of the font because it creates too many edge cases and undesirable output.

**Comment by Shedward** [comment]
I actually want to add to this topic. Those ligatures can create a problems in some languages where things like [Error] or [error] have some meaning.

For example in Swift I might want to use an array of errors with code like this
```swift
func errorsExample() -> [Error] {
    var errors: [Error] = []

    if let error = makeSingleError() {
        errors = [error]
    } else {
        errors = makeMultipleErrors()
    }
    return errors
}
```

which with Pragmata Pro will create this output

<img width="277" alt="Screenshot 2020-05-14 at 12 05 26" src="https://user-images.githubusercontent.com/904623/81915624-829e2480-95db-11ea-9301-6861c4f58793.png">
 
Not only it just looks weird, it also erases difference between type and value.

@fabrizioschiavi Is there any way to manually rebuild font without this feature by default?
I didn't find a way to configure open type features in Xcode.

**Comment by ElonPark** [comment]
> I actually want to add to this topic. Those ligatures can create a problems in some languages where things like [Error] or [error] have some meaning.
> 
> For example in Swift I might want to use an array of errors with code like this
> 
> ```swift
> func errorsExample() -> [Error] {
>     var errors: [Error] = []
> 
>     if let error = makeSingleError() {
>         errors = [error]
>     } else {
>         errors = makeMultipleErrors()
>     }
>     return errors
> }
> ```
> 
> which with Pragmata Pro will create this output
> 
> <img alt="Screenshot 2020-05-14 at 12 05 26" width="277" src="https://user-images.githubusercontent.com/904623/81915624-829e2480-95db-11ea-9301-6861c4f58793.png">
> 
> Not only it just looks weird, it also erases difference between type and value.
> 
> @fabrizioschiavi Is there any way to manually rebuild font without this feature by default?
> I didn't find a way to configure open type features in Xcode.

How about this

```swift
func errorsExample() -> Array<Error> {
    var errors = Array<Error>()

    if let singleError = makeSingleError() {
        errors = [singleError]
    } else {
        errors = makeMultipleErrors()
    }
    return errors
}
```

**Comment by fabrizioschiavi** [comment]
Wow! Thank you friends, I must adjust it absolutely. Ligatures can't become an obstacle to coding.

**Comment by nomasprime** [comment]
Originally thought these ligatures were a nice idea so I raised an issue to get them working in iTerm: alphanumeric ligatures weren't rendered for performance reasons.

But, like I said before, these alphanumeric ligatures just have too many edge cases and are generally too problematic for serious programmers. If you're willing to mainain another variant, a compromise would be to have two ligature variants: one standard and another standard plus alphanumeric ligatures.

Personally I don't use ligatures at all anymore:

- All the above
- They make motions, especially in Vim, unnecessarily difficult
- General performance cost too high

I can understand why standard ligatures might be nice or useful for specialised use cases but these alphanumeric ligatures just seem superfluous to me.

PS Thank you @fabrizioschiavi for such a great font 👏
